jsapp/js/searches.es6

```diff
@@ -115,16 +118,19 @@ function SearchContext(opts={}) {
     },
     // remove asset from all search store lists
     removeAsset(asset) {
-      this.removeAssetFromList(asset.uid, 'defaultQueryResultsList');
-      this.removeAssetFromList(asset.uid, 'searchResultsList');
-      this.rebuildCategorizedList(
-        this.state.defaultQueryResultsList,
-        'defaultQueryCategorizedResultsLists'
-      );
-      this.rebuildCategorizedList(
-        this.state.searchResultsList,
-        'searchResultsCategorizedResultsLists'
-      );
+      // only update things if given asset matches the current context types
+      if (this.state.defaultQueryFilterParams?.assetType.includes(asset.asset_type)) {
```

Comment:
Just a heads up, this may cause a conflict with: https://github.com/kobotoolbox/kpi/blob/17cd0bb21de7c9683f6754f4417f188d76f81d39/jsapp/js/searches.es6#L119-L122 (PR #2984)

Reply:
@magicznyleszek, a FYI, didn't notice this until after the fact, but when you delete an asset form the asset list, the callback gives you `asset.assetType` instead of the `asset.asset_type` that it gives you when you create an asset.  I've fixed it with https://github.com/kobotoolbox/kpi/commit/b2022fa7c61c3adde9b86db331d5c986f701f54a.

Reply:
Awesome :pray: